### PR TITLE
Fix for #PyDev 1020: Raw f-strings (and b-strings) not recognized if r is first

### PR DESCRIPTION
**Pull request by oluiscabral**

### CODE REVIEW

File: plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar38/python.jjt [hunk 1]
```diff
@@ -421,11 +421,19 @@ MORE : /* Strings */
 |   < (["b", "B"]) (["r", "R"])? "\"" > :  IN_BSTRING21
 |   < (["b", "B"]) (["r", "R"])? "'''" > :  IN_BSTRING13
 |   < (["b", "B"]) (["r", "R"])? "\"\"\"" > :  IN_BSTRING23
+|   < (["r", "R"] ["b", "B"])? "'" > :  IN_BSTRING11
```
Comment (fabioz) [inline plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar38/python.jjt:424]: Seems good, the only thing is that these changes now have to be ported to python.jjt_template (and then, run make_replace.py to update this file).

File: plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java [hunk 2]
```diff
@@ -481,9 +481,23 @@ public void makeString(Token t, int quotes, Str strToFill) {
...
+                    fstring = true;
+                    start++;
+                } else if (quoteCharAux == 'b' || quoteCharAux == 'B') {
+                    bstring = true;
+                    start++;
+                }
+            }
+            String str;
+            if (bstring) {
+                str = 'b' + s.substring(quotes + start, s.length() - quotes + 1);
```
Comment (fabioz) [inline plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java:496]: Humm, why is this done here? I don't think we need special treatment for the str here...

Comment (oluiscabral) [reply plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java]: bstrings in Python 3.8 always output with a standard style ( " b'{foo}' " )

Comment (oluiscabral) [reply plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java]: Python 3.8 output bstring in " b'xxxxx' " style

File: plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java [hunk 3]
```diff
@@ -141,4 +141,690 @@ public void testPositionalArgs4() {
         parseLegalDocStr("a = lambda a, b=1, /, c=2, d=3, *, e=4, f=5: None");
     }
 
+    public void testUString() throws Exception {
```
Comment (fabioz) [inline plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:144]: The tests don't seem consistent with what was already there... I checked locally and the expected value inside the string shouldn't really include the prefixes.

I'm providing a function below which simplifies the testing a bit (please add the remaining use cases, but note that something as `ur'foo'` returns `foo` as the value, thus, something as `b'foo` should also return just `foo` -- the `Str` ast attributes should be checked for that).

Also, you don't need that many tests (just a few for the corner cases as empty strings/ string with contents) should be ok.

```
    private void checkStr(String s, String expected, boolean binary, boolean fstring, boolean unicode, boolean raw) {
        SequencialASTIteratorVisitor visitor = SequencialASTIteratorVisitor.create(parseLegalDocStr(s));
        List<ASTEntry> asList = visitor.getAsList(Str.class);
        assertEquals(asList.size(), 1);
        ASTEntry entry = asList.iterator().next();
        Str str = (Str) entry.node;
        assertEquals(binary, str.binary);
        assertEquals(fstring, str.fstring);
        assertEquals(unicode, str.unicode);
        assertEquals(raw, str.raw);
        assertEquals(expected, str.s);
    }

    public void testStr() throws Exception {
        checkStr("''", "", false, false, false, false);
        checkStr("'s'", "s", false, false, false, false);

        checkStr("b''", "", true, false, false, false);
        checkStr("b's'", "s", true, false, false, false);

        checkStr("f''", "", false, true, false, false);
        checkStr("f's'", "s", false, true, false, false);
    }
```

File: plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java (hunk 2 above)
Comment (fabioz) [reply plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/DefaultPythonGrammarActions.java]: I checked it here and it's not really consistent with others, so, it shouldn't be done.

File: plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java [hunk 4]
```diff
@@ -141,4 +146,27 @@ public void testPositionalArgs4() {
...
+
+    public void testStr() throws Exception {
+        checkStr("''", "", false, false, false, false);
+        checkStr("'s'", "s", false, false, false, false);
+
+        checkStr("b''", "", true, false, false, false);
+        checkStr("b's'", "s", true, false, false, false);
+
+        checkStr("f''", "", false, true, false, false);
+        checkStr("f's'", "s", false, true, false, false);
```
Comment (fabioz) [inline plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:170]: It's missing the tests for what was just added. i.e.: `rf'xxx'` and `rb'xxx'`.

Comment (fabioz) [reply plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:170]: (i.e.: my previous comment was just about how to structure the tests, not all the tests that should be done -- it doesn't need to be exhaustive but it should at least check for the new features).

Comment (oluiscabral) [reply plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:176]: Thats right.. I completely passed by it. My bad

Comment (fabioz) [reply plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:176]: One thing that I've just checked is that you can also use `br` and not just `rb`. Can you check that too?

Comment (fabioz) [reply plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java:176]: Oops, it's actually there.
